src/senselab/utils/tasks/dimensionality_reduction.py

```diff
@@ -1,4 +1,60 @@
-"""This module provides the implementation of dimensionality reduction."""
+"""This module provides the implementation of dimensionality reduction.
+
+Dimensionality reduction is a crucial preprocessing step in many machine learning and data analysis pipelines.
+It aims to project a high-dimensional dataset into a lower-dimensional subspace while preserving as much of the
+significant structure and information as possible. This process is particularly useful for visualization,
+noise reduction, and improving computational efficiency in downstream tasks.
+
+This module specifically implements three popular dimensionality reduction techniques: Principal Component Analysis
+(PCA), t-Distributed Stochastic Neighbor Embedding (t-SNE), and Uniform Manifold Approximation and Projection (UMAP).
+
+### Task Overview
+- Input: High-dimensional data (typically represented as a matrix or tensor), i.e. A tensor of shape (100, 10)
+- Output: Lower-dimensional representation of the input data, i.e. A tensor of shape (100, 2)
+- Goal: Preserve important structures or relationships in the data while reducing its dimensionality
+
+### Evaluation Metrics
+The effectiveness of dimensionality reduction techniques can be evaluated using several metrics, including but not
+limited to:
+- **Reconstruction Error**: Measures the difference between the original data and its reduced representation when
+    projected back to the original space. Commonly used for techniques like PCA.
+- **Trustworthiness**: Assesses how much the model preserves neighbor relationships between the high-dimensional
+    space and the low-dimensional space.
+- **Continuity**: Evaluates how well the local structure of the data is preserved in the reduced space.
+
+### Supported models
```

Comment:
way better, would be great to have some refs

Reply:
Sounds good, I've added links to the t-SNE and UMAP papers, as well as a Nature primer on PCA that I found.